nifi-commons/nifi-expression-language/src/main/java/org/apache/nifi/attribute/expression/language/evaluation/functions/Uuid3Evaluator.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.attribute.expression.language.evaluation.functions;
+
+import org.apache.nifi.attribute.expression.language.EvaluationContext;
+import org.apache.nifi.attribute.expression.language.evaluation.Evaluator;
+import org.apache.nifi.attribute.expression.language.evaluation.QueryResult;
+import org.apache.nifi.attribute.expression.language.evaluation.StringEvaluator;
+import org.apache.nifi.attribute.expression.language.evaluation.StringQueryResult;
+
+import java.nio.ByteBuffer;
+import java.util.Objects;
+import java.util.UUID;
+import org.apache.commons.lang3.ArrayUtils;
+
+public class Uuid3Evaluator extends StringEvaluator {
+
+    private final Evaluator<String> subject;
+    private final Evaluator<String> namespace;
+
+    public Uuid3Evaluator(final Evaluator<String> subject, final Evaluator<String> namespace) {
+        this.subject = subject;
+        this.namespace = namespace;
+    }
+
+    @Override
+    public QueryResult<String> evaluate(final EvaluationContext evaluationContext) {
+        final String subjectValue = subject.evaluate(evaluationContext).getValue();
+        final String nsValue = namespace.evaluate(evaluationContext).getValue();
+        final UUID nsUUID = UUID.fromString(Objects.requireNonNull(nsValue));
+
+        final byte[] nsBytes =
+            ByteBuffer.wrap(new byte[16])
+                .putLong(nsUUID.getMostSignificantBits())
+                .putLong(nsUUID.getLeastSignificantBits())
+                .array();
+
+        final byte[] subjectBytes = Objects.requireNonNull(subjectValue).getBytes();
```

Comment:
I think this is throwing a NullRef when the attribute referenced doesn't exist

Reply:
Fixed. Returns `new StringQueryResult(null)` now if the subject is null.